x-pack/plugins/observability_solution/observability_ai_assistant/server/service/knowledge_base_service/recall_from_search_connectors.ts

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import { IUiSettingsClient } from '@kbn/core-ui-settings-server';
+import { isEmpty, orderBy, compact } from 'lodash';
+import type { Logger } from '@kbn/logging';
+import { CoreSetup } from '@kbn/core-lifecycle-server';
+import { firstValueFrom } from 'rxjs';
+import { RecalledEntry } from '.';
+import { aiAssistantSearchConnectorIndexPattern } from '../../../common';
+import { ObservabilityAIAssistantPluginStartDependencies } from '../../types';
+
+export async function recallFromSearchConnectors({
+  queries,
+  esClient,
+  uiSettingsClient,
+  logger,
+  core,
+}: {
+  queries: Array<{ text: string; boost?: number }>;
+  esClient: { asCurrentUser: ElasticsearchClient; asInternalUser: ElasticsearchClient };
+  uiSettingsClient: IUiSettingsClient;
+  logger: Logger;
+  core: CoreSetup<ObservabilityAIAssistantPluginStartDependencies>;
+}): Promise<RecalledEntry[]> {
+  const connectorIndices = await getConnectorIndices(esClient, uiSettingsClient, logger);
+  logger.debug(`Found connector indices: ${connectorIndices}`);
+
+  const [semanticTextConnectors, legacyConnectors] = await Promise.all([
```

Comment:
These requests happen in parallel. We could make it a single bulk request but I don't think it matters too much